src/covid_model_seiir_pipeline/regression/task.py

```diff
@@ -1,68 +1,74 @@
-                                                          draw_id=draw_id)
+    location_data = data_interface.load_infection_data(draw_id, location_ids)
+    location_data = {location_id: location_data[location_data['location_id'] == location_id].copy()
+                     for location_id in location_ids}
     covariates = data_interface.load_covariates(regression_specification.covariates, location_ids)
 
-    # Run ODE fit
+    logger.info('Preparing inputs.')
     np.random.seed(draw_id)
     beta_fit_inputs = model.ODEProcessInput(
         df_dict=location_data,
-        col_date=INFECTION_COL_DICT['COL_DATE'],
-        col_cases=INFECTION_COL_DICT['COL_CASES'],
-        col_pop=INFECTION_COL_DICT['COL_POP'],
-        col_loc_id=INFECTION_COL_DICT['COL_LOC_ID'],
-        col_lag_days=INFECTION_COL_DICT['COL_ID_LAG'],
-        col_observed=INFECTION_COL_DICT['COL_OBS_DEATHS'],
+        col_date='date',
+        col_infections='infections_draw',
+        col_pop='population',
+        col_loc_id='location_id',
+        col_lag_days='duration',
+        col_observed='observed_deaths',
         alpha=regression_specification.parameters.alpha,
         sigma=regression_specification.parameters.sigma,
         gamma1=regression_specification.parameters.gamma1,
         gamma2=regression_specification.parameters.gamma2,
         solver_dt=regression_specification.parameters.solver_dt,
         day_shift=regression_specification.parameters.day_shift,
     )
+
+    logger.info('Modeling past compartments data from infections.')
     ode_model = model.ODEProcess(beta_fit_inputs)
     beta_fit = ode_model.process()
     beta_fit['date'] = pd.to_datetime(beta_fit['date'])
 
     # Run regression
+    logger.info('Running regression.')
     mr_data = model.align_beta_with_covariates(covariates, beta_fit, list(regression_specification.covariates))
     regressor = model.build_regressor(regression_specification.covariates.values())
     coefficients = regressor.fit(mr_data)
     log_beta_hat = compute_beta_hat(covariates, coefficients)
     beta_hat = np.exp(log_beta_hat).rename('beta_pred').reset_index()
 
     # Format and save data.
+    logger.info('Formatting and writing outputs.')
     data_df = pd.concat(location_data.values())
-    data_df['date'] = pd.to_datetime(data_df['date'])
     regression_betas = beta_hat.merge(covariates, on=['location_id', 'date'])
     regression_betas = beta_fit.merge(regression_betas, on=['location_id', 'date'], how='left')
-    merged = data_df.merge(regression_betas, on=['location_id', 'date'], how='outer').sort_values(['location_id', 'date'])
-    merged = merged[(merged['obs_deaths'] == 1) | (merged['deaths_draw'] > 0)]
+    merged = (data_df
+              .merge(regression_betas, on=['location_id', 'date'], how='outer')
+              .sort_values(['location_id', 'date']))
+    merged = merged[(merged['observed_deaths'] == 1) | (merged['deaths_draw'] > 0)]
```

Comment:
I _think_ @JosephWagner would write this as `merged = merged.loc[..., :]`. I'm still not clear why to use one over the other.

Or maybe I'm just wrong about Joe

Reply:
That's a thing I would do.  I tend to use .loc and .iloc for everything.  Not mucking with logic here though.